spec.bs

```diff
@@ -276,17 +276,109 @@ href="https://html.spec.whatwg.org/multipage/embedded-content-other.html#dimensi
+
+  : <dfn>nested config mapping</dfn>
+  :: a [=map=] whose [=map/keys=] are [=urn uuids=] and whose [=map/values=] are [=fenced frame configs=]
+</dl>
+
+Each [=fenced frame config mapping=] has a <dfn for="fenced frame config mapping" export>maximum
+number of configs</dfn>, which is implementation-defined. The [=fenced frame config mapping/maximum
+number of configs=] may be a nonnegative number or infinity.
+
+Note: It is important to specify the behavior of
+[=fenced frame config mapping/maximum number of configs=] because its semantics can interact with
+config-generating APIs in a privacy sensitive way.
+
+At a high level, in order to store a [=fenced frame config=] in the
+[=traversable navigable/fenced frame config mapping=], one must first store a pending config, and
+then turn the pending config into a finalized config. Those procedures are as follows:
+
+<div algorithm>
+  To <dfn for="fenced frame config mapping" export>store a pending config</dfn> into a particular
+  [=fenced frame config mapping=] given a [=fenced frame config=] |config|, run these steps:
+
+  1. Let |pendingMapping| be the [=fenced frame config mapping/pending config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If the [=map/size=] of the [=fenced frame config mapping/pending config mapping=] + the
+     size of the [=fenced frame config mapping/finalized config mapping=] meets or
+     exceeds the [=fenced frame config mapping/maximum number of configs=], return failure.
+
+  1. Let |urn| be a randomly generated [=urn uuid=].
+
+  1. [=map/Set=] |pendingMapping|[|urn|] to |config|.
+</div>
+
+<div algorithm>
+  To <dfn export>finalize a pending config</dfn> in a particular [=fenced frame config mapping=]
+  given a [=urn uuid=] |urn|, run these steps:
+
+  1. Let |pendingMapping| be the [=fenced frame config mapping/pending config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. Let |finalizedMapping| be the [=fenced frame config mapping/finalized config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If |pendingMapping|[|urn|] does not [=map/exist=], return failure.
+
+  1. Let |finalizedConfig| be |pendingMapping|[|urn|].
+
+  1. [=map/Remove=] |pendingMapping|[|urn|].
+
+  1. [=map/Set=] |finalizedMapping|[|urn|] be |finalizedConfig|.
+</div>
+
+<div algorithm>
+  To <dfn export>store nested configs</dfn> into a particular [=fenced frame config mapping=] given
+  a [=fenced frame config/nested configs=] |nestedConfigs|, run these steps:
+
+  1. Let |nestedMapping| be the [=fenced frame config mapping/nested config mapping=] of the
+     associated [=fenced frame config mapping=].
+
+  1. If |nestedConfigs| is null, return.
+
+  1. [=map/iterate|For each=] |urn| → |config| of |nestedConfigs|:
+
+     1. [=map/Set=] |nestedMapping|[|urn|] to |config|.
+</div>
+
+
+<div algorithm>
+  In order to <dfn export>look up a [=fenced frame config=] at a [=urn uuid=] in the
```

Comment:
Can you follow the algorithm trend I did in https://github.com/WICG/fenced-frame/pull/67/commits/321f2a0a10304bba50790c8cc75c064a89a2fa0f with this one, especially regarding usage of infra primitives?